pkg/helper/flowcollector.go

```diff
@@ -99,6 +83,139 @@ func UseLoki(spec *flowslatest.FlowCollectorSpec) bool {
 	return spec.Loki.Enable == nil || *spec.Loki.Enable
 }
 
```

Comment:
In this whole file, I think instead of having getter functions for every single field, in that case, it would be simpler to create a new struct (that would be very similar to the Manual struct.. maybe derived from it) and have conversion functions to this struct for every mode.
I can do a suggestion around that if this isn't clear...

Reply:
Why not. That's a good candidate for the followup with `monolith` and `distributed` modes.